eventuals/lazy.h

```diff
@@ -12,17 +12,20 @@ namespace eventuals {
 ////////////////////////////////////////////////////////////////////////
 
 template <typename T, typename... Args>
-class _Lazy final {
+class Lazy final {
  public:
   template <typename Tuple>
-  _Lazy(Tuple args)
+  Lazy(Tuple args)
```

Comment:
Can you see if we can delete this one and instead always pass the 'args' and construct the tuple from that?